services/matrix/matrix.service.js

```diff
@@ -166,36 +184,56 @@ export default class Matrix extends BaseJsonService {
     } else {
       host = serverFQDN
     }
-    const accessToken = await this.retrieveAccessToken({ host })
-    const lookup = await this.lookupRoomAlias({ host, roomAlias, accessToken })
-    const data = await this._requestJson({
-      url: `https://${host}/_matrix/client/r0/rooms/${encodeURIComponent(
-        lookup.room_id,
-      )}/state`,
-      schema: matrixStateSchema,
-      options: {
-        searchParams: {
-          access_token: accessToken,
+    if (host.toLowerCase() === 'matrix.org' || fetchMode === 'summary') {
```

Comment:
There is quite a lot going on in this one function now.

Could we split this chunk down into 2 functions: Lets say `fetchSummary()` and `fetchGuest()`.

Reply:
Done 52253bd40ba8fec26e0ff03d5edff6b08ad7aeca